docs/tutorial_advanced.rst

```diff
@@ -693,26 +693,40 @@ This attribute accepts a string that corresponds to the name of the container pl
+The default command that the container runs can be overwritten by setting the :attr:`command <reframe.core.containers.ContainerPlatform.command>` of the container platform.
+
+The :attr:`image <reframe.core.containers.ContainerPlatform.image>` is the only mandatory attribute for container-based checks.
 It is important to note that the :attr:`executable <reframe.core.pipeline.RegressionTest.executable>` and :attr:`executable_opts <reframe.core.pipeline.RegressionTest.executable_opts>` attributes of the actual test are ignored in case of container-based tests.
 
 ReFrame will run the container as follows:
 
 .. code-block:: console
 
-    singularity exec -B"/path/to/test/stagedir:/workdir" docker://ubuntu:18.04 bash -c 'cd rfm_workdir; pwd; ls; cat /etc/os-release'
+    singularity exec -B"/path/to/test/stagedir:/rfm_workdir" --pwd="/rfm_workdir" docker://ubuntu:18.04 bash -c 'pwd; cat /etc/os-release > release.txt'
 
-By default ReFrame will mount the stage directory of the test under ``/rfm_workdir`` inside the container and it will always prepend a ``cd`` command to that directory.
-The user commands are then run from that directory one after the other.
+By default ReFrame will mount the stage directory of the test under ``/rfm_workdir`` inside the container.
 Once the commands are executed, the container is stopped and ReFrame goes on with the sanity and performance checks.
-Users may also change the default mount point of the stage directory by using :attr:`workdir <reframe.core.pipeline.RegressionTest.container_platform.workdir>` attribute:
 Besides the stage directory, additional mount points can be specified through the :attr:`mount_points <reframe.core.pipeline.RegressionTest.container_platform.mount_points>` attribute:
 
 .. code-block:: python
 
     self.container_platform.mount_points = [('/path/to/host/dir1', '/path/to/container/mount_point1'),
                                             ('/path/to/host/dir2', '/path/to/container/mount_point2')]
 
+
+The container filesystem is ephemeral, therefore, ReFrame mounts the stage directory under ``/rfm_workdir`` inside the container where the user can copy directories/files as needed.
+The aforementioned directories/files will then be available inside the stage directory after the container execution finishes.
+This is very useful if the above directories/files are going to be used for the sanity/performance checks.
+If this is the case, the user should overwrite the default command executed by the container, using the :attr:`command <reframe.core.containers.ContainerPlatform.command>` to include the appropriate copy commands.
+In this test, we have changed the directory that the container is going to run the commands using the `--pwd=/rfm_workdir` through the :attr:`options <reframe.core.containers.ContainerPlatform.options>`.
```

Comment:
I would prefer if you add the `cp` as you just proposing above. It can be confusing.

Reply:
Do you mean to add also the command I use to copy the artifacts?

Reply:
I am suggesting to not use the `--pwd`. Let the container execute on its ephemeral filesystem and then copy the  `releases.txt` to the `/rfm_workdir`, as you are exactly suggesting right above.